.github/workflows/test-move-prover.yml

```diff
@@ -1,38 +1,78 @@
 name: Test_Move_Prover
 
 on:
+  pull_request:
+  # types: [opened, reopened, synchronize]
   repository_dispatch:
     types: [new_PR]
 
 jobs:
   install_and_test_Move_Prover:
 
-    runs-on: ubuntu-latest
-
+    runs-on: ubuntu-18.04
+  
     steps:
+    - name: Checkout libra
+      uses: actions/checkout@v2
+      with:
+        path: "libra"
+        repository: ${{ github.event.client_payload.owner }}/${{ github.event.client_payload.repo }} # repo of the forked libra/libra
+        ref: ${{ github.event.client_payload.pull_ref }} # commit that triggered the PR
+    - name: Detect the changes in move-prover
+      working-directory: ./libra
+      run: |
+        git fetch origin
+        git diff --name-only origin/master
+        git diff --name-only origin/master | grep "language/move-prover"
```

Comment:
will this work if the PR didn't create a new branch and is working on master?

you might want to diff against the base instead ([see how I get the base here](https://github.com/mimoo/mirai-bot/blob/master/.github/workflows/rust-mirai.yml#L46:L51))